### DISCUSSION TITLE
Epoch wise training instead of steps based for UNETR

### DISCUSSION THREAD

**Discussion by umgpy** [Project-MONAI/MONAI#3801] [Q&A]
Hi @ahatamiz,

In the UNETR tutorial, is there a reason that training is done with steps rather than epoch wise. I have tried converting it to regular epoch wise training but it is slower that the original method (possibly because of my faulty implementation).

 Just wondering if this a best case practice for training transformer based networks.

Thanks in advance

**Comment by ahatamiz** [comment]
Hi @umgpy, 

Thanks for your interest in our work. Our initial code used an iteration-based approach and hence the tutorial followed the same logic. However, our studies showed that both epoch-based and iteration-based training of UNETR yield similar results. In MONAI research repository, we provide a fast and efficient epoch-based version for UNETR training on BTCV dataset which also supports multi-gpu training. I recommend checking it here:

https://github.com/Project-MONAI/research-contributions/tree/master/UNETR/BTCV

Thanks,
